### PR TITLE
feat: redesign image preview and optimize some ux/ui, add modal rc

### PR DESCRIPTION
**Pull request by Innei**
# Main Changes

1. Optimize Swiper Image, click to open in a new window, and click the title bar to open URL
2. Unique images
3. Scoped swiper style in component
4. Empty status UI

# Updated

1. Modal component
2. Image Preview Hook

# Note

Modal Component with two ways to use. Refer: https://github.com/Innei/rc-modal https://innei.in/posts/tech/why-i-prefer-imperative-modal

### CODE REVIEW

**Comment by DIYgod** [comment]
1. The empty icon is misaligned.

image

2. Opening a new window by clicking the image is not ideal, as it forces users to switch applications. Maybe we can use a popup instead?

3. Placing the swiper buttons on the right side might confuse users about whether it controls the left or right image.

**Comment by Innei** [comment]
CleanShot 2024-06-06 at 10  16 10@2x
CleanShot 2024-06-06 at 10  16 24@2x

**Comment by rebron1900** [comment]
三个图标大小也许该一致

**Comment by Innei** [comment]
https://discord.com/channels/1243823539426033696/1243823761913020458/1248513666412843088

https://discord.com/channels/1243823539426033696/1243823761913020458/1248547618804727849

**Review by DIYgod — APPROVED** [review]

File: src/renderer/src/store/unread.ts [hunk 1]
```diff
@@ -12,26 +12,34 @@ interface UnreadActions {
...
     const { data } = unread
-    for (const feedId in data) {
-      set((state) =>
-        produce(state, (state) => {
+    get().internal_reset()
+    set((state) =>
+      produce(state, (state) => {
+        for (const feedId in data) {
           state.data[feedId] = data[feedId]
           return state
```
Comment (DIYgod) [inline src/renderer/src/store/unread.ts:38]: This line is redundant

File: src/renderer/src/store/unread.ts [hunk 2]
```diff
@@ -12,26 +12,34 @@ interface UnreadActions {
...
-    )
+    const unread = await apiClient.reads.$get({
+      query: { view: String(view) },
+    })
 
     const { data } = unread
-    for (const feedId in data) {
-      set((state) =>
-        produce(state, (state) => {
+    get().internal_reset()
```
Comment (DIYgod) [inline src/renderer/src/store/unread.ts:33]: This line of code seems to cause the unread count to be rendered as 0

**Pull request merged**